### PR TITLE
Llm testing live run

### PR DESCRIPTION
**Pull request by nv-braf**
Changes required to get a live run working.
I now have a complete profile running successfully through PA!

### CODE REVIEW

File: model_analyzer/config/generate/brute_plus_binary_search_run_config_generator.py [hunk 1]
```diff
@@ -116,7 +116,11 @@ def _create_brute_run_config_generator(self) -> BruteRunConfigGenerator:
 
     def _can_binary_search_top_results(self) -> bool:
         for model in self._models:
-            if model.parameters()["concurrency"] or model.parameters()["request_rate"]:
+            if (
+                model.parameters()["concurrency"]
+                or model.parameters()["request_rate"]
+                or self._config.is_llm_model()
```
Comment (nv-hwoo) [inline model_analyzer/config/generate/brute_plus_binary_search_run_config_generator.py:122]: What is this if statement checking and will it ever be false?

File: model_analyzer/config/generate/perf_analyzer_config_generator.py [hunk 2]
```diff
@@ -424,7 +448,7 @@ def _create_parameter_perf_config_values(self) -> dict:
 
         if self._cli_config.is_llm_model():
             perf_config_values["request-parameter"] = [
-                "max_token:" + str(mtc) + ":int" for mtc in self._max_token_counts
+                "max_tokens:" + str(mtc) + ":int" for mtc in self._max_token_counts
```
Comment (nv-hwoo) [inline model_analyzer/config/generate/perf_analyzer_config_generator.py:451]: f-string?
```suggestion
                f"max_tokens:{mtc}:int" for mtc in self._max_token_counts
```

File: model_analyzer/perf_analyzer/perf_analyzer.py [hunk 3]
```diff
@@ -539,7 +547,7 @@ def _calculate_avg_first_token_latency(self, llm_output: Dict) -> float:
                 request["response_timestamps"][0] - request["timestamp"]
             )
 
-        avg_first_token_latency = mean(total_first_token_latencies)
+        avg_first_token_latency = float(mean(total_first_token_latencies))
```
Comment (nv-hwoo) [inline model_analyzer/perf_analyzer/perf_analyzer.py:550]: We can use `np.mean`. I think it returns float as default.

File: model_analyzer/perf_analyzer/perf_analyzer.py [hunk 4]
```diff
@@ -554,7 +562,7 @@ def _calculate_avg_token_to_token_latency(self, llm_output: Dict) -> float:
 
             token_to_token_latencies.append(mean(response_to_response_latencies))
 
-        avg_token_to_token_latency = mean(token_to_token_latencies)
+        avg_token_to_token_latency = float(mean(token_to_token_latencies))
```
Comment (nv-hwoo) [inline model_analyzer/perf_analyzer/perf_analyzer.py:565]: See above.

File: tests/common/test_utils.py [hunk 5]
```diff
@@ -300,9 +306,15 @@ def construct_perf_analyzer_config(
 
     if llm_search_mode:
         pa_config._args["request-parameter"] = (
-            "max_token:" + str(max_token_count) + ":int"
+            "max_tokens:" + str(max_token_count) + ":int"
```
Comment (nv-hwoo) [inline tests/common/test_utils.py:309]: f-string?
```suggestion
            f"max_tokens:{max_token_count}:int"
```

File: tests/common/test_utils.py [hunk 6]
```diff
@@ -300,9 +306,15 @@ def construct_perf_analyzer_config(
...
         )
-        pa_config._args["input-data"] = DEFAULT_INPUT_JSON_PATH + "/input-data.json"
+        pa_config._args["request-period"] = (
+            max_token_count if max_token_count < 10 else 10
+        )
+        pa_config._args["input-data"] = (
+            DEFAULT_INPUT_JSON_PATH + "/input-data-" + str(text_input_length) + ".json"
```
Comment (nv-hwoo) [inline tests/common/test_utils.py:315]:
```suggestion
            DEFAULT_INPUT_JSON_PATH + f"/input-data-{text_input_length}.json"
```

File: model_analyzer/config/generate/perf_analyzer_config_generator.py [hunk 7]
```diff
@@ -377,8 +377,24 @@ def _extract_text_input_length(
...
+            )
+
         perf_config.update_config(parameter_combination)
 
+    def _extract_max_token_from_request_parameter(self, request_parameter: str) -> int:
+        max_token_start = request_parameter.find(":")
+        max_token_stop = request_parameter.find(":", max_token_start + 1)
+        max_token = int(request_parameter[max_token_start + 1 : max_token_stop])
+
+        return max_token
```
Comment (nv-hwoo) [inline model_analyzer/config/generate/perf_analyzer_config_generator.py:396]: I think `split` could be useful here
```suggestion
        _, value, _ = request_parameter.split(":")
        return int(value)
```

File: model_analyzer/config/generate/perf_analyzer_config_generator.py [hunk 8]
```diff
@@ -377,8 +377,24 @@ def _extract_text_input_length(
     def _update_perf_config_based_on_parameter_combination(
         self, perf_config: PerfAnalyzerConfig, parameter_combination: Dict
     ) -> None:
+        if "request-parameter" in parameter_combination:
+            request_parameter = parameter_combination["request-parameter"]
+            max_token = self._extract_max_token_from_request_parameter(
```
Comment (nv-hwoo) [inline model_analyzer/config/generate/perf_analyzer_config_generator.py:382]: `max_token` -> `max_tokens` for consistency
```suggestion
            max_tokens = self._extract_max_tokens_from_request_parameter(
```

File: model_analyzer/config/generate/perf_analyzer_config_generator.py [hunk 9]
```diff
@@ -377,8 +377,24 @@ def _extract_text_input_length(
...
+                request_parameter
+            )
+            parameter_combination["request-period"] = (
+                max_token if max_token < 10 else 10
```
Comment (nv-hwoo) [inline model_analyzer/config/generate/perf_analyzer_config_generator.py:386]: Is there reason why we are capping request period to 10?

File: model_analyzer/config/generate/perf_analyzer_config_generator.py [hunk 10]
```diff
@@ -400,21 +417,28 @@ def _update_perf_config_for_llm_model(
...
 
-        perf_config.update_config({"input-data": self._input_json_filename})
+        perf_config.update_config({"input-data": input_json_filename})
 
     def _modify_text_in_input_dict(self, text_input_length: int) -> Dict:
         modified_text = " ".join(repeat("Hello", text_input_length))
 
         modified_input_dict = {k: v for k, v in self._llm_input_dict.items()}
-        modified_input_dict["data"][0]["text-input"] = modified_text
+        modified_input_dict["data"][0]["PROMPT"] = [modified_text]
```
Comment (nv-hwoo) [inline model_analyzer/config/generate/perf_analyzer_config_generator.py:434]: Let's put FIXME so we don't forget to change this back to `text_input`

File: model_analyzer/perf_analyzer/perf_analyzer.py [hunk 11]
```diff
@@ -285,6 +285,14 @@ def _get_single_model_cmd(self, index):
         if self._is_multi_model():
             cmd += ["--enable-mpi"]
         cmd += self._get_pa_cli_command(index).replace("=", " ").split()
+
+        # OPTME: There should be a more elegant way of determining how to add EOS
+        #        We have to do it here because we use a dictionary to create the PA command
+        #        and it already contains `--request-parameter`
+        if "--periodic-concurrency-range" in cmd:
+            cmd.append("--request-parameter")
+            cmd.append("ignore_eos:true:bool")
```
Comment (nv-hwoo) [inline model_analyzer/perf_analyzer/perf_analyzer.py:294]: Were you able to make any progress with the link that @matthewkotila gave you?

File: model_analyzer/perf_analyzer/perf_analyzer.py (hunk 3 above)
Comment (nv-braf) [reply model_analyzer/perf_analyzer/perf_analyzer.py:550]: I am using `np.mean` it returns a `float64`, but we're expecting `float` so I'm converting it.

File: model_analyzer/perf_analyzer/perf_analyzer.py (hunk 4 above)
Comment (nv-braf) [reply model_analyzer/perf_analyzer/perf_analyzer.py:565]: See above

File: model_analyzer/config/generate/brute_plus_binary_search_run_config_generator.py (hunk 1 above)
Comment (nv-braf) [reply model_analyzer/config/generate/brute_plus_binary_search_run_config_generator.py:122]: It's determining if we should be a binary search of the concurrency/request-rate space for our top configurations. I'm disabling this for LLM models as it's not relevant.

File: model_analyzer/config/generate/perf_analyzer_config_generator.py (hunk 10 above)
Comment (nv-braf) [reply model_analyzer/config/generate/perf_analyzer_config_generator.py:451]: Done

File: model_analyzer/config/generate/perf_analyzer_config_generator.py (hunk 9 above)
Comment (nv-braf) [reply model_analyzer/config/generate/perf_analyzer_config_generator.py]: Good catch. This is stale code before I added the RCS option for `request-period`. I've updated the code to use `request-period`

File: model_analyzer/perf_analyzer/perf_analyzer.py (hunk 11 above)
Comment (nv-braf) [reply model_analyzer/perf_analyzer/perf_analyzer.py:294]: This is a larger change than I want to do right now. It'd be nice to clean this up, but not necessary at this point.

**Pull request merged**